pkg/lldp/client.go

```diff
@@ -69,7 +69,10 @@ func (l *Client) Start(resultChan chan<- DiscoveryResult) {
 					if layer.LayerType() != layers.LayerTypeLinkLayerDiscoveryInfo {
 						continue
 					}
-					info := layer.(*layers.LinkLayerDiscoveryInfo)
+					info, ok := layer.(*layers.LinkLayerDiscoveryInfo)
+					if !ok {
```

Comment:
So `layer.LayerType() != layers.LayerTypeLinkLayerDiscoveryInfo` check doesn't work?

Reply:
might be, i dont know, but this is a part in the code where this goroutine could fail, at least it makes it better.

Reply:
The comparison above compares layer.LayerType with `LayerTypeLinkLayerDiscoveryInfo`, and later it is TypeAsserted to `LinkLayerDiscoveryInfo` which is obviosly not the same